### PR TITLE
add fb pixel code to base registration template

### PR DESCRIPTION
**Pull request by biyeun**
buddy @proteusvacuum 

needs to be merged today

### CODE REVIEW

File: corehq/apps/registration/templates/registration/base.html [hunk 1]
```diff
@@ -2,6 +2,21 @@
...
 {% block head %}{{ block.super }}
     {% include 'analytics/fullstory.html' %}
+    <!-- Facebook Pixel Code -->
+    <script>
+    !function(f,b,e,v,n,t,s){if(f.fbq)return;n=f.fbq=function(){n.callMethod?
+    n.callMeyythod.apply(n,arguments):n.queue.push(arguments)};if(!f._fbq)f._fbq=n;
+    n.push=n;n.loaded=!0;n.version='2.0';n.queue=[];t=b.createElement(e);t.async=!0;
+    t.src=v;s=b.getElementsByTagName(e)[0];s.parentNode.insertBefore(t,s)}(window,
+    document,'script','https://connect.facebook.net/en_US/fbevents.js');
+    fbq('init', '301081893674207');
```
Comment (proteusvacuum) [inline corehq/apps/registration/templates/registration/base.html]: Is this a user id / password? If so, it should probably go into a settings file.
We also shouldn't be including this on enikshay / icds / other enterprise sites, right?

**Review by proteusvacuum — CHANGES_REQUESTED** [review]

File: corehq/apps/registration/templates/registration/base.html (hunk 1 above)
Comment (biyeun) [reply corehq/apps/registration/templates/registration/base.html]: yep. I just copied the code from an email. I'll go ahead and update. good catch

Comment (proteusvacuum) [reply corehq/apps/registration/templates/registration/base.html]: should probably get a new ID now that this will be in our git history.

**Comment by biyeun** [comment]
@proteusvacuum amended for git history erasure 😳

File: corehq/apps/registration/templates/registration/base.html [hunk 2]
```diff
@@ -2,6 +2,23 @@
...
+    !function(f,b,e,v,n,t,s){if(f.fbq)return;n=f.fbq=function(){n.callMethod?
+    n.callMeyythod.apply(n,arguments):n.queue.push(arguments)};if(!f._fbq)f._fbq=n;
+    n.push=n;n.loaded=!0;n.version='2.0';n.queue=[];t=b.createElement(e);t.async=!0;
+    t.src=v;s=b.getElementsByTagName(e)[0];s.parentNode.insertBefore(t,s)}(window,
+    document,'script','https://connect.facebook.net/en_US/fbevents.js');
+    fbq('init', '{{ FACEBOOK_PIXEL_ID }}');
+    fbq('track', 'PageView');
+    </script>
+    <noscript><img height="1" width="1" style="display:none"
+    src="https://www.facebook.com/tr?id=301081893674207&ev=PageView&noscript=1"
```
Comment (proteusvacuum) [inline corehq/apps/registration/templates/registration/base.html]: This looks like the same ID that was in the `init` bit before

**Review by proteusvacuum — CHANGES_REQUESTED** [review]
Unfortunately, you can't just `git amend` this out, as the commit still exists: https://github.com/dimagi/commcare-hq/commit/0bcba4096211767e92939ee89b1ed53ba1132382

I think you might have to actually change the ID in facebook.

File: corehq/apps/registration/templates/registration/base.html [hunk 3]
```diff
@@ -2,6 +2,23 @@
 
 {% block head %}{{ block.super }}
     {% include 'analytics/fullstory.html' %}
+    {% if FACEBOOK_PIXEL_ID %}
```
Comment (gcapalbo) [inline corehq/apps/registration/templates/registration/base.html]: this needs to be loaded into the template context right?

File: corehq/apps/registration/templates/registration/base.html [hunk 4]
```diff
@@ -2,6 +2,23 @@
...
+    <!-- Facebook Pixel Code -->
+    <script>
+    !function(f,b,e,v,n,t,s){if(f.fbq)return;n=f.fbq=function(){n.callMethod?
+    n.callMeyythod.apply(n,arguments):n.queue.push(arguments)};if(!f._fbq)f._fbq=n;
```
Comment (proteusvacuum) [inline corehq/apps/registration/templates/registration/base.html]: this is weird, but is this actually `callMeyythod`? 
The pixel docs seem to say this should be `callMethod`.

File: corehq/apps/registration/templates/registration/base.html (hunk 3 above)
Comment (proteusvacuum) [inline corehq/apps/registration/templates/registration/base.html]: Does this need to be defined somewhere?

**Review by proteusvacuum — CHANGES_REQUESTED** [review]

File: corehq/apps/registration/templates/registration/base.html (hunk 3 above)
Comment (biyeun) [reply corehq/apps/registration/templates/registration/base.html]: there is middleware that handles this. I tested it locally

File: corehq/apps/registration/templates/registration/base.html (hunk 4 above)
Comment (biyeun) [reply corehq/apps/registration/templates/registration/base.html]: I copied and pasted this code from comms. what about `n.callMethod?` right before? Looks like legacy code

Comment (biyeun) [reply corehq/apps/registration/templates/registration/base.html]: ...as in. I didn't consult the docs for this. I assume Comms got it from a generator.

File: corehq/apps/registration/templates/registration/base.html (hunk 3 above)
Comment (biyeun) [reply corehq/apps/registration/templates/registration/base.html]: middleware takes care of it

**Comment by biyeun** [comment]
@proteusvacuum also this https://github.com/dimagi/commcarehq-ansible/pull/955

File: corehq/apps/registration/templates/registration/base.html (hunk 3 above)
Comment (biyeun) [reply corehq/apps/registration/templates/registration/base.html]: https://github.com/dimagi/commcarehq-ansible/pull/955

Comment (biyeun) [reply corehq/apps/registration/templates/registration/base.html]: there appears to be middleware taking care of this, as `GOOGLE_ANALYTICS_API_ID` is not defined anywhere in a template context, yet is accessed the same way from a template.

Comment (biyeun) [reply corehq/apps/registration/templates/registration/base.html]: I also tested locally and it works.

File: corehq/apps/registration/templates/registration/base.html (hunk 4 above)
Comment (calellowitz) [reply corehq/apps/registration/templates/registration/base.html]: @biyeun I think Farid is correct. That looks like a copy paste error on my part. Not sure how it got there, but its not in the original version carrie sent

Comment (biyeun) [reply corehq/apps/registration/templates/registration/base.html]: added it to the default blank localsettings.py dict so people know it's available.

**Comment by gcapalbo** [comment]
isn't this FACEBOOK_PIXEL_ID going to be loaded on the registration page (which is public facing)? how come we need to add it to localsettings?

**Comment by proteusvacuum** [comment]
sorry for all the comments @biyeun ... did you mean to do a submodule update in the last commit?

**Comment by biyeun** [comment]
@gcapalbo that's a great question :joy: that's why I didn't think much about it when I just copy pasted the code...
I guess for best practices. but having the ID in there allows you to selectively show it only when it's present in localsettings, so enikshay for instance won't see this.

**Comment by biyeun** [comment]
@proteusvacuum omg 😳

**Comment by biyeun** [comment]
@proteusvacuum i'm so done with this PR. this change should have taken 5 mins, and now i'm diving into ansible :joy: 🙄

**Comment by gcapalbo** [comment]
i think we should get into the habit of using settings.ENTERPRISE_MODE to filter this stuff out

**Comment by biyeun** [comment]
@proteusvacuum @calellowitz confirmed that somewhere down the line someone may have mistakenly typed some `yy`s.....

**Comment by biyeun** [comment]
@gcapalbo but what if enikshay wanted some facebook tracking? 🙄

**Comment by biyeun** [comment]
note: decided to go with @gcapalbo suggestion to use `settings.SERVER_ENVIRONMENT == 'production'` and skip the ansible shenanigans. THANK YOU. 🙏

**Review by proteusvacuum — APPROVED** [review]
sorry for the back and forth.

**Comment by biyeun** [comment]
@proteusvacuum all valid points tho, and you caught the `callMeyythod` :joy: :rofl: maybe it should have been typoed to `calMeyythod` 😜  @calellowitz

**Pull request merged**